lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameServerImpl.java

```diff
@@ -75,8 +79,10 @@ private NameServerImpl(
     final Codec<NamingMessage> codec = NamingCodecFactory.createFullCodec(factory);
     final EventHandler<NamingMessage> handler = createEventHandler(codec);
 
-    injector.bindVolatileParameter(RemoteConfiguration.HostAddress.class, localAddressProvider.getLocalAddress());
+    String host = UNKNOWN_HOST_NAME.equals(hostAddress) ? localAddressProvider.getLocalAddress() : hostAddress;
```

Comment:
I'm concerned why this is necessary. Shouldn't the LocalAddressProvider be abstracting this logic away?

Reply:
No. None of the LocalAddressProvider implementations abstract this. This is similar to how it is in NettyMessagingTransport.

---
In reply to: [199967859](https://github.com/sharathmalladi/reef/pull/6#discussion_r199967859) [](ancestors = 199967859)